localstack-core/localstack/services/cloudformation/v2/provider.py

```diff
@@ -448,6 +454,33 @@ def describe_change_set(
         )
         return result
 
+    @handler("DeleteChangeSet")
+    def delete_change_set(
+        self,
+        context: RequestContext,
+        change_set_name: ChangeSetNameOrId,
+        stack_name: StackNameOrId = None,
+        **kwargs,
+    ) -> DeleteChangeSetOutput:
+        state = get_cloudformation_store(context.account_id, context.region)
+
+        if is_changeset_arn(change_set_name):
+            change_set = state.change_sets.get(change_set_name)
+        elif not is_changeset_arn(change_set_name) and stack_name:
+            change_set = find_change_set_v2(state, change_set_name, stack_name)
```

Comment:
We need to early return if the change set cannot be found

```suggestion
            change_set = find_change_set_v2(state, change_set_name, stack_name)
            if change_set is None:
                # AWS does not raise an error if it cannot find the change set
                return
```